test/boost/repair_test.cc

```diff
@@ -133,3 +136,102 @@ SEASTAR_TEST_CASE(flush_repair_rows_on_wire_to_sstable) {
+                ? this_shard_id()
+                : tests::random::get_int(remote_sharder.shard_count() - 1);
+            const auto random_range = std::invoke([&] {
+                const auto left = tests::random::get_int(token_min, max_token);
+                const auto right = tests::random::get_int(left == token_max ? token_max : left + 1, token_max);
+                return dht::token_range::make(
+                    {dht::token::from_int64(left), tests::random::with_probability(0.5)},
+                    {dht::token::from_int64(right), tests::random::with_probability(0.5)});
+            });
+            auto read_all = [&](repair_reader::read_strategy strategy) -> future<std::vector<mutation_fragment>> {
+                auto reader = repair_reader(e.db(), cf, cf.schema(), make_reader_permit(e),
+                    random_range, remote_sharder, remote_shard, 0, strategy);
+                std::vector<mutation_fragment> result;
+                while (auto mf = co_await reader.read_mutation_fragment()) {
+                    result.push_back(std::move(*mf));
+                }
+                co_await reader.on_end_of_stream();
+                co_await reader.close();
+                co_return result;
+            };
+            auto dump = [&](std::ostream& target, const std::vector<mutation_fragment>& fragments) {
+                for (const auto& f: fragments) {
+                    ::fmt::print(target, "\n{}", mutation_fragment::printer(s, f));
+                }
+            };
+
+            const auto data1 = co_await read_all(strategy1);
+            const auto data2 = co_await read_all(strategy2);
+            std::optional<sstring> mismatch;
+            if (data1.size() != data2.size()) {
+                mismatch = ::format("size1 {} != size2 {}", data1.size(), data2.size());
+            } else {
+                for (int i = 0; i < data1.size(); ++i) {
+                    const auto& mf1 = data1[i];
+                    const auto& mf2 = data2[i];
+                    if (!mf1.equal(s, mf2)) {
+                        mismatch = ::format("{} != {}",
+                            mutation_fragment::printer(s, mf1), mutation_fragment::printer(s, mf2));
+                        break;
+                    }
+                }
+            }
+            if (mismatch) {
+                std::cout << "data1:";
+                dump(std::cout, data1);
+                std::cout << "\ndata2:";
+                dump(std::cout, data2);
+                std::cout << std::endl;
+                BOOST_FAIL(::format("s1={}, s2={}, mismatch={}", strategy1, strategy2, *mismatch));
+            }
+        };
+
+        co_await do_check(dht::sharder(local_sharder.shard_count() + 1, local_sharder.sharding_ignore_msb()),
+            repair_reader::read_strategy::multishard_split,
+            repair_reader::read_strategy::multishard_filter);
+        co_await do_check(local_sharder,
```

Comment:
How about adding one more check for local  vs split

```    
    co_await do_check(local_sharder,
            repair_reader::read_strategy::local,
            repair_reader::read_strategy::multishard_split);
```

Reply:
local vs split
local vs filter 

are the two we use in practice.

Reply:
added, please approve